libcxx/include/__ranges/concat_view.h

```diff
@@ -0,0 +1,626 @@
+                ix, [&](auto index_constant) {
+                  constexpr size_t index_x = index_constant.value;
+                  auto dx = ranges::__distance(ranges::begin(std::get<index_x>(x.parent_->views_)), it_x);
+
+                  it_y.apply_fn_with_const_index<std::tuple_size_v<decltype(y.parent_->views_)>>(
+                      iy, [&](auto index_constant) {
+                        constexpr size_t index_y = index_constant.value;
+                        auto dy = ranges::__distance(ranges::begin(std::get<index_y>(y.parent_->views_)), it_y);
+                        difference_type s = 0;
+                        for (size_t idx = index_y + 1; idx < index_x; idx++) {
+                          s += ranges::size(std::get<idx>(x.parent_->views_));
+                        }
+                        return dy + s + dx;
+                      });
+                });
+          },
+          x.it_,
+          y.it_);
+    } else if (ix < iy) {
+      return -(y - x);
+    } else {
+      std::visit([&](const auto& it1, const auto& it2) { return it1 - it2; }, x.it_, y.it_);
+    }
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr iterator operator-(const iterator& it, difference_type n)
+    requires __concat_is_random_access<_Const, _Views...>
+  {
+    auto temp = it;
+    temp -= n;
+    return temp;
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr difference_type operator-(const iterator& x, default_sentinel_t)
+    requires(sized_sentinel_for<sentinel_t<__maybe_const<_Const, _Views>>, iterator_t<__maybe_const<_Const, _Views>>> &&
+             ...) &&
+            (__apply_drop_first<_Const, _Views...>::value)
+  {
+    size_t ix = x.it_.index();
+    std::visit(
+        [&](auto& it_x) {
+          it_x.apply_fn_with_const_index<std::tuple_size_v<decltype(x.parent_->views_)>>(ix, [&](auto index_constant) {
+            constexpr size_t index_x = index_constant.value;
+            auto dx                  = ranges::__distance(ranges::begin(std::get<index_x>(x.parent_->views_)), it_x);
+
+            difference_type s = 0;
+            for (size_t idx = 0; idx < index_x; idx++) {
+              s += ranges::size(std::get<idx>(x.parent_->views_));
+            }
+
+            return -(dx + s);
+          });
+        },
+        x.it_);
+  }
+
+  _LIBCPP_HIDE_FROM_ABI friend constexpr difference_type operator-(default_sentinel_t, const iterator& x)
+    requires(sized_sentinel_for<sentinel_t<__maybe_const<_Const, _Views>>, iterator_t<__maybe_const<_Const, _Views>>> &&
+             ...) &&
+            (__apply_drop_first<_Const, _Views...>::value)
+  {
+    -(x - default_sentinel);
+  }
+};
+
+namespace views {
+namespace __concat {
+struct __fn {
+  template <class... _Views>
+  [[nodiscard]] _LIBCPP_HIDE_FROM_ABI constexpr auto operator()(_Views... views) const
+      noexcept(concat_view(std::forward<_Views>(views)...)) -> decltype(concat_view(std::forward<_Views>(views)...)) {
```

Comment:
Missed `noexcept` operator.
```suggestion
      noexcept(noexcept(concat_view(std::forward<_Views>(__views)...))) -> decltype(concat_view(std::forward<_Views>(views)...)) {
```